### ISSUE TITLE
connection arguments not validated before execution

### ISSUE THREAD

**Issue by sheerun** [hashicorp/terraform#10620]
Hi there,

Terraform v0.7.13

I'm iterating on my terraform file and accidentally wrote `azurerm_network_interface.master.ip_address` instead of `azurerm_public_ip.master.ip_address`

Terraform happily started executed plan, and failed only after creating vm.

### Affected Resource(s)

- connection

### Terraform Configuration Files

```hcl
# ...

resource "azurerm_network_interface" "master" {
  name = "${var.prefix}-master-interface"
  location = "${var.location}"
  resource_group_name = "${azurerm_resource_group.default.name}"
  network_security_group_id = "${azurerm_network_security_group.default.id}"

  ip_configuration {
    name = "${var.prefix}-ipconfig"
    subnet_id = "${azurerm_subnet.dmz.id}"
    private_ip_address_allocation = "dynamic"
    public_ip_address_id = "${azurerm_public_ip.master.id}"
  }
}

resource "azurerm_virtual_machine" "master" {
  name = "${var.prefix}-master"

  location = "${var.location}"
  resource_group_name = "${azurerm_resource_group.default.name}"
  network_interface_ids = ["${azurerm_network_interface.master.id}"]
  vm_size = "Standard_A0"

  delete_os_disk_on_termination = true
  delete_data_disks_on_termination = true

  storage_image_reference {
    publisher = "Canonical"
    offer = "UbuntuServer"
    sku = "16.04.0-LTS"
    version = "latest"
  }

  storage_os_disk {
    name = "master"
    vhd_uri = "${azurerm_storage_account.adafs.primary_blob_endpoint}${azurerm_storage_container.images.name}/master.vhd"
    caching = "ReadWrite"
    create_option = "FromImage"
  }

  os_profile {
    computer_name = "master"
    admin_username = "ubuntu"
    admin_password = "***"
  }

  os_profile_linux_config {
    disable_password_authentication = true
    ssh_keys = [
      {
        path = "/home/ubuntu/.ssh/authorized_keys",
        key_data = "${var.keys}"
      }
    ]
  }

  tags {
    environment = "staging"
  }

  connection {
    host = "${azurerm_network_interface.master.ip_address}"
    user     = "ubuntu"
    private_key = "${var.private_key}"
  }

  provisioner "remote-exec" {
    inline = [
      "sudo apt update",
      "sudo apt upgrade -y",
      "curl -fsSL https://test.docker.com/ | sudo sh"
    ]
  }
}
```

### Debug Output

```
Error applying plan:

1 error(s) occurred:

* Resource 'azurerm_network_interface.master' does not have attribute 'ip_address' for variable 'azurerm_network_interface.master.ip_address'
```

### Expected Behavior
It should validate that I entered inexisting attribute and fail *before* executing plan (during validation), not during executin, leaving server in invalid state.

### Actual Behavior
It started executing plan, created vm, and then errored with above message

**Comment by mitchellh** [comment]
Yep, there is another issue tracking this and we plan to fix this soon, in <= 0.9. Thanks!
